libminifi/src/core/ProcessSession.cpp

```diff
@@ -199,21 +198,19 @@ void ProcessSession::remove(const std::shared_ptr<core::FlowFile> &flow) {
   flow->setDeleted(true);
   deleted_flowfiles_.push_back(flow);
   std::string reason = process_context_->getProcessorNode()->getName() + " drop flow record " + flow->getUUIDStr();
-  provenance_report_->drop(flow, reason);
+  provenance_report_->drop(*flow, reason);
 }
 
-void ProcessSession::putAttribute(const std::shared_ptr<core::FlowFile>& flow, std::string_view key, const std::string& value) {
-  flow->setAttribute(key, value);
-  std::stringstream details;
-  details << process_context_->getProcessorNode()->getName() << " modify flow record " << flow->getUUIDStr() << " attribute " << key << ":" << value;
-  provenance_report_->modifyAttributes(flow, details.str());
+void ProcessSession::putAttribute(core::FlowFile& flow_file, std::string_view key, const std::string& value) {
+  flow_file.setAttribute(key, value);
+  std::string details = fmt::format("{} modify flow record {} attribute {} : {}", process_context_->getProcessorNode()->getName(), flow_file.getUUIDStr(), key, value);
```

Comment:
there was no space between the key and the value before the change
```suggestion
  std::string details = fmt::format("{} modify flow record {} attribute {}:{}", process_context_->getProcessorNode()->getName(), flow_file.getUUIDStr(), key, value);
```

Reply:
good catch fixed it in https://github.com/apache/nifi-minifi-cpp/pull/1707/commits/d64e7f890b82be04ec5a75bc484a1ebcbb001f32